### PR TITLE
add downsamplings

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: gnomad_constraint/resources/resource_utils.py [hunk 1]
```diff
@@ -360,6 +360,25 @@ def get_constraint_tsv_path(
...
+
+    :param version: One of the release versions (`VERSIONS`). Default is
+        `CURRENT_VERSION`.
+    :param test: Whether the Table is for testing purposes. Default is False.
+    :return: TSV path of constraint metrics.
+    """
+    check_param_scope(version=version)
+
+    return (
+        f"{get_constraint_root(version, test)}/metrics/tsv/gnomad.v{version}.downsampling_constraint_metrics.txt.bgz"
```
Comment (jkgoodrich) [inline gnomad_constraint/resources/resource_utils.py:378]:
```suggestion
        f"{get_constraint_root(version, test)}/metrics/tsv/gnomad.v{version}.downsampling_constraint_metrics.tsv.bgz"
```

File: gnomad_constraint/utils/constraint.py [hunk 2]
```diff
@@ -459,18 +465,21 @@ def apply_models(
...
+        keep_annotations=keep_annotations,
+        pops=pops,
+        downsamplings=downsamplings,
+        grouping=grouping,
+        partition_hint=obs_pos_count_partition_hint,
         filter_coverage_over_0=True,
         transcript_for_synonymous_filter=None,
     )
 
+    ht = ht.checkpoint(new_temp_file(prefix="oe_ht", extension="ht"))
```
Comment (jkgoodrich) [inline gnomad_constraint/utils/constraint.py:481]: Add a comment about why this checkpoint it needed/helpful

File: gnomad_constraint/utils/constraint.py [hunk 3]
```diff
@@ -531,9 +556,11 @@ def apply_models(
     ht = ht.annotate_globals(
         apply_model_params=hl.struct(
             max_af=max_af,
-            pops=pops,
+            genetic_ancestry_groups=pops,
             plateau_models=plateau_models,
             coverage_model=coverage_model_global,
+            high_cov_definition=high_cov_definition,
```
Comment (jkgoodrich) [inline gnomad_constraint/utils/constraint.py:562]: nice

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
Just a few small things

File: gnomad_constraint/utils/constraint.py (hunk 2 above)
Comment (klaricch) [reply gnomad_constraint/utils/constraint.py:481]: nm, it's checkpointed in the function

File: gnomad_constraint/pipeline/constraint_pipeline.py [hunk 4]
```diff
@@ -242,17 +246,21 @@ def main(args):
...
     custom_vep_annotation = args.custom_vep_annotation
     gerp_lower_cutoff = args.gerp_lower_cutoff
     gerp_upper_cutoff = args.gerp_upper_cutoff
 
-    pops = constraint_res.POPS if use_pops else ()
-
     if version not in constraint_res.VERSIONS:
         raise ValueError("The requested version of resource Tables is not available.")
 
+    # If "global" is the only population specified for v4 and above, use the pared-down downsampling list.
```
Comment (jkgoodrich) [inline gnomad_constraint/pipeline/constraint_pipeline.py:258]:
```suggestion
    # If "global" is the only population specified for v4, use the pared-down downsampling list.
```

**Review by jkgoodrich — APPROVED** [review]
tiny nit pick, but LGTM!

**Pull request merged**